src/api/app/assets/javascripts/webui2/airbrake-js.js.erb

```diff
@@ -0,0 +1,25 @@
+//= require airbrake-js-client
+var errbit_id = "<%= CONFIG['errbit_javascript_api_key'] %>";
+var errbit_host = "<%= CONFIG['errbit_javascript_host'] %>";
+
+if (errbit_id) {
+  var airbrake = new airbrakeJs.Client({
+    projectId: 1,
+    projectKey: errbit_id,
+    host: errbit_host,
+    environment: 'production',
+  });
+
+  window.onerror = function (message, file, line, col, error) {
+    var promise = airbrake.notify(error);
+    promise.then(function(notice) {
+      if (notice.id) {
+        // console.log('Notified errbit. Notice:', notice.id);
+        console.log('Notified errbit.');
+      } else {
+        // console.log('Notifying errbit failed. Reason:', notice.error);
+        console.log('Notifying errbit failed!');
```

Comment:
Shouldn't we let the exception bubble up if the errbit notification failed? We're swallowing the error trace otherwise right?

Reply:
No this after the error and it's processing. See the [window.onerror](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror) docu.

![Screenshot from 2019-08-29 12-50-37](https://user-images.githubusercontent.com/514785/63934459-aa545900-ca5b-11e9-9525-3d31557c9a2d.png)

Reply:
And users can't really fix the error if something goes wrong when sending it to errbit.

Reply:
> No this after the error and it's processing. See the [window.onerror](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror) docu.
> 
> ![Screenshot from 2019-08-29 12-50-37](https://user-images.githubusercontent.com/514785/63934459-aa545900-ca5b-11e9-9525-3d31557c9a2d.png)

Alright! :+1:

Reply:
> And users can't really fix the error if something goes wrong when sending it to errbit.

You are right, my concerns were more on the line of the devs trying to debug stuff that does not work and this swallowing the error traces, but as you posted before, everything is ok, so :coffee: :dog: :fire: